### ISSUE TITLE
DMX effects not working / Strange Sequence Save Behavior

### ISSUE THREAD

**Issue by nmiller0113** [smeighan/xLights#2895]
**Describe the bug**
I'm attaching a packaged sequence that I used last year. I adjusted the timing of a single DMX effect on the fogmach model and then re-saved. After I save the sequence I get a popup when closing xLights that asked me to save my changes, which I already had, but it doesn't seem like it takes. Then if I save it a second time it doesn't say anything. I can recreate this bug over and over. On top of that, and most importantly, the DMX effect is no longer outputting in FPP even though a test from FPP will trigger the DMX output, and other sequences which I did not change are still working and outputting to the DMX interface as expected. No channels have changed from last year. I'm only attaching one sequence exhibiting this behavior, but there are others that I modified that are doing the same thing. I can also create a brand new sequence and recreate these issues.

**To Reproduce**
Described above. Please let me know if you need more information.

**Expected behavior**
Sequence shouldn't prompt me to save even after I've saved, and DMX should output in FPP.

**Screenshots**
<img width="550" alt="Screen Shot 2021-11-01 at 10 25 58 PM" src="https://user-images.githubusercontent.com/1048868/139792252-9cbd7b24-79bf-4da7-b749-fa3e12d6c472.png">

**Versions (please complete the following information):**
 - OS: Mac 11.6.1
 - xLights version 2021.34

**Attachments**
[Creepshow.zip](https://github.com/smeighan/xLights/files/7458108/Creepshow.zip)

**Comment by nmiller0113** [comment]
I've confirmed this behavior is the same with 2021.35 as well. I'm not sure if anyone has looked at this yet, but hopefully that info is helpful! Thank you!

**Comment by nmiller0113** [comment]
Any thoughts on this one? I've tried different things and cannot get my DMX to output from the rendered FSEQ. Just to mention again what I said when I opened the case. The old sequences still work. It's only any new sequence or when I edit an old sequence that it doesn't play the DMX output. Plus it forces me to save twice, but still doesn't make a difference. The sequence attached to this case is a new one so you should hopefully be able to determine what the issue is. Thanks again and sorry to bother on this...I just want to get the DMX output working again :)

**Comment by patdelaney** [comment]
Can you provide more information on what the FPP is?  (Pi or Beagle) and what hat or cape?

**Comment by nmiller0113** [comment]
@patdelaney, I think we can leave FPP out of this as a variable (I'll change the bug title to remove confusion). I've Here's why I say this:

- It plays all my old/non-modified sequences fine and DMX outputs fine. One more thing to note on these. I recreated all of my FSEQ's for my old sequences in this version of xLights and they all output great
- This issue only happens if I modify an old sequence or create a new one. And then, when I modify or create it always does the strange behavior of telling me I didn't save the sequence...even when I literally had just saved it and prior to trying to close xLights. If I hit the save button for the second time, the message does not return when closing the sequence...yet the issue of the sequence actually working still exists

This really feels like an issue in the creation of the FSEQ and something in xLights getting confused with the Layout model and the effect in use. I've been using xLights in this exact same setup for 5+ years and never had an issue. Something is being buggy no doubt :)

Just to answer your question even though I'm 100% confident it has nothing to do with my issue, I'm running the latest version of FPP on a Pi, and it's streaming DDP to an F48v4 with this DMX device connected directly to it. I can run tests and trigger the DMX without issue....and as I mentioned earlier, the old/non-modified sequence work all the time...even with new FSEQ's created in xLights. Something is definitely strange about this.

**Comment by nmiller0113** [comment]
@patdelaney - I just discovered something else that may be of significance. I realized that the strange save behavior only exhibits itself when add/removing/moving/changing the DMX effect on the channel block model. If I add the DMX effect to any other model in the sequence it works fine, and if I add any other effect to the channel block model it works fine. Which makes sense since my DMX device depends on both the channel block model and DMX effect to work together and this seems to be the only place the issue exists.

I can provide a video to reflect this, but I'm fairly certain that you'll be able to reproduce it with this information and the sequence I've attached...which is the same sequence I'm testing with.

**Comment by keithsw1111** [comment]
I fixed the unnecessary save prompt. The rest I cannot reproduce. Does it play correctly directly from xLights. If so then it is a config problem between xLights/FPP ... it is unlikely it is the FSEQ.

**Comment by nmiller0113** [comment]
@keithsw1111 thank you for correcting the save issue but I’m unsure of how the other issue could be configuration related. I’m using the exact same model and DMX effect on all my sequences, old and new, and it works on the old sequences that have not been modified and doesn’t on any of the new ones. I’ll test again tomorrow but I’ve tested this numerous times now and it’s the same every time. Also, I recreated all of my FSEQ’s so I’d imagine that if I had a config issue it would impact all of them.

I’ll try playing them directly from xLights to my f48v4 tomorrow and eliminate FPP in the middle. Thanks again!

**Comment by nmiller0113** [comment]
@keithsw1111 I tried it from xLights directly to my f48 and it worked fine. I then turned my attention to FPP and noticed that the files were adding some strange symbols to the name when running FPP connect from time to time. I assumed something had corrupted in the file system or the SD Card was going bad. So I reflashed it and the SD Card wouldn't even complete a wipe. I flashed a brand new SD Card and then brought FPP back online. Now all is well. Sorry that I bothered for what turned out to be a bad SD Card.

Thank you for correcting the unnecessary save prompt! I'll leave this case open until you mark it with the merge of that fix. Thank you again!
